scripts/checks.sh

```diff
@@ -439,6 +439,46 @@ function check_service_restarts()
 	fi
 }
 
+function check_image_corruption()
+{
+	local images output result="${GOOD}"
+	local corrupted=()
+	local timeout=()
+	local -i corrupted_count timeout_count
+	# TODO: this command is filtering out any images with a size <1Kb (for delta-based images)
```

Comment:
I'm unsure of usual practice at Balena. The `TODO` seems done (implemented in the next line); is this a marker for documentation, or optimization, or future refactoring?

Also, is it worth breaking out the next line to its own function?  It's a small thing, but seeing a line like:

```
images=$(get_images_larger_than_1kb)
```

or some such might help readability, and make it easier to rewrite/refactor if needed.

Reply:
The TODO is hanging around because I don't really like that filter, but can't think of a cleaner mechanism to use (future refactoring improvement). We could use what robertgzr mentioned above, but not all devices use delta v3 and we have to support back to balenaOS v2.11.0.

Reply:
Makes sense to me.  Again, unsure what usual practice is; if you figure it's worth adding that context to the note, then :+1: .